src/utils/installPackage.js

```diff
@@ -72,19 +74,22 @@ async function installPeerDependencies(filepath, name, options) {
 }
 
 async function determinePackageManager(filepath) {
-  let configFile = await config.resolve(filepath, [
-    'yarn.lock',
-    'package-lock.json'
-  ]);
-  let hasYarn = await checkForYarnCommand();
-
-  // If Yarn isn't available, or there is a package-lock.json file, use npm.
-  let configName = configFile && path.basename(configFile);
-  if (!hasYarn || configName === 'package-lock.json') {
+  const yarnLockFile = await config.resolve(filepath, [YARN_LOCK]);
+
+  /**
+   * no yarn.lock => use npm
+   * yarn.lock => Use yarn, fallback to npm
+   */
+  if (!yarnLockFile) {
```

Comment:
It would've been shorter as 

```Javascript
if (yarnLockFile && (await checkForYarnCommand())) {
	return 'yarn';
}
return 'npm';
```

But I merged it anyway, it's just a minor thing anyway